File: template-server/components/typographyContent.go
package components

import graphql "github.com/hasura/go-graphql-client"

type TypographyContent struct {
	Paragraph []struct {
		Typename graphql.String `graphql:"__typename" json:"__typename,omitempty"`
		Value    graphql.String `json:"value,omitempty"`
	} `json:"paragraph,omitempty"`
}

File: template-server/src/fragments/typography-content.ts
import { gql } from "graphql-request";
import { paragraphFragment } from "./paragraph";

const typographyContentFragment = gql`
  fragment typographyContentFragment on TypographyContent {
    __typename
    paragraph {
      ...paragraphFragment
    }
  }
  ${paragraphFragment}
`;

export { typographyContentFragment };